src/client/serialize/daos_serialize.c

```diff
@@ -0,0 +1,1026 @@
+		H5Tclose(usr_attr_val_vtype);
+	if (file_id > 0)
+		H5Fclose(file_id);
+	return rc;
+}
+
+static int
+deserialize_str(hid_t file_id, struct daos_prop_entry *entry,
+		const char *prop_str)
+{
+	hid_t	status = 0;
+	int	rc = 0;
+	hid_t	attr_dtype = 0;
+	hid_t	cont_attr = 0;
+	size_t	buf_size;
+
+	cont_attr = H5Aopen(file_id, prop_str, H5P_DEFAULT);
+	if (cont_attr < 0) {
+		rc = -DER_MISC;
+		D_ERROR("failed to open property attribute "DF_RC"\n",
+			DP_RC(rc));
+		D_GOTO(out, rc);
+	}
+	attr_dtype = H5Aget_type(cont_attr);
+	if (attr_dtype < 0) {
+		rc = -DER_MISC;
+		D_ERROR("failed to open property attribute type "DF_RC"\n",
+			DP_RC(rc));
+		D_GOTO(out, rc);
+	}
+	buf_size = H5Tget_size(attr_dtype);
+	if (buf_size <= 0) {
+		rc = -DER_MISC;
+		D_ERROR("failed to get size for property attribute "DF_RC"\n",
+			DP_RC(rc));
+		D_GOTO(out, rc);
+	}
+	D_ALLOC(entry->dpe_str, buf_size);
+	if (entry->dpe_str == NULL) {
+		rc = -DER_MISC;
+		D_ERROR("failed to allocate property attribute "DF_RC"\n",
+			DP_RC(rc));
+		D_GOTO(out, rc);
+	}
+	status = H5Aread(cont_attr, attr_dtype, entry->dpe_str);
+	if (status < 0) {
+		rc = -DER_IO;
+		D_ERROR("failed to read property attribute %s "DF_RC"\n",
+			entry->dpe_str, DP_RC(rc));
+		D_GOTO(out, rc);
+	}
+out:
+	if (cont_attr > 0)
+		H5Aclose(cont_attr);
+	if (attr_dtype > 0)
+		H5Tclose(attr_dtype);
+	return rc;
+}
+
+static int
+deserialize_uint(hid_t file_id, struct daos_prop_entry *entry,
+		 const char *prop_str)
+{
+	hid_t   status = 0;
+	int     rc = 0;
+	hid_t   cont_attr = 0;
+	hid_t   attr_dtype = 0;
+
+	cont_attr = H5Aopen(file_id, prop_str, H5P_DEFAULT);
+	if (cont_attr < 0) {
+		rc = -DER_MISC;
+		D_ERROR("failed to open property attribute "DF_RC"\n",
+			DP_RC(rc));
+		D_GOTO(out, rc);
+	}
+	attr_dtype = H5Aget_type(cont_attr);
+	if (attr_dtype < 0) {
+		rc = -DER_MISC;
+		D_ERROR("failed to open property attribute type "DF_RC"\n",
+			DP_RC(rc));
+		D_GOTO(out, rc);
+	}
+	status = H5Aread(cont_attr, attr_dtype, &entry->dpe_val);
+	if (status < 0) {
+		rc = -DER_MISC;
+		D_ERROR("failed to read property attribute "DF_RC"\n",
+			DP_RC(rc));
+		D_GOTO(out, rc);
+	}
+out:
+	if (cont_attr > 0)
+		H5Aclose(cont_attr);
+	if (attr_dtype > 0)
+		H5Tclose(attr_dtype);
+	return rc;
+}
+
+static int
+deserialize_acl(hid_t file_id, struct daos_prop_entry *entry,
+		const char *prop_str)
+{
+	hid_t		status = 0;
+	int		rc = 0;
+	int		ndims = 0;
+	const char      **rdata = NULL;
+	struct daos_acl	*acl;
+	htri_t		acl_exist;
+	hid_t		cont_attr = 0;
+	hid_t		attr_dtype = 0;
+	hid_t		attr_dspace = 0;
+	hsize_t		attr_dims[1];
+
+	/* First check if the ACL * attribute exists. */
+	acl_exist = H5Aexists(file_id, prop_str);
+	if (acl_exist < 0) {
+		/* Actual error  */
+		rc = -DER_MISC;
+		D_ERROR("failed to open property attribute type "DF_RC"\n",
+			DP_RC(rc));
+		D_GOTO(out, rc);
+	} else if (acl_exist == 0) {
+		/* Does not exist, but that's okay. */
+		rc = 0;
+		D_GOTO(out, rc);
```

Comment:
This is also possible
```suggestion
		D_GOTO(out, rc = 0);
```